src/bentoml/_internal/server/metrics/prometheus.py

```diff
@@ -108,59 +135,343 @@ def generate_latest(self):
+
+        Gauge also provide ``track_inprogress()``, to track inprogress object.
+        This function can also be used as either a context manager or a decorator.
+
+        .. tab-set::
+
+           .. tab-item:: Example
+
+              .. code-block:: python
+
+                 from bentoml.metrics import Gauge
+
+                 g = Gauge('inprogress_request', 'Request inprogress')
+
+                 @svc.api(input=JSON(), output=JSON())
+                 @g.track_inprogress()
+                 def predict(input_data: dict[str, str]):
+                     ...
+
+           .. tab-item:: Context Manager
+
+              .. code-block:: python
+
+                 from bentoml.metrics import Gauge
+
+                 g = Gauge('inprogress_request', 'Request inprogress')
+
+                 @svc.api(input=JSON(), output=JSON())
+                 def predict(input_data: dict[str, str]):
+                     with g.track_inprogress():
+                         ...
+
+                The gauge will increment when the context is entered and decrement when the context is exited.
+
+        Args:
+            name (str): The name of the metric.
+            documentation (str): A documentation string.
+            labelnames (tuple[str]): A tuple of strings specifying the label names for the metric. Defaults to ``()``.
+            namespace (str): The namespace of the metric. Defaults to an empty string.
+            subsystem (str): The subsystem of the metric. Defaults to an empty string.
+            unit (str): The unit of the metric. Defaults to an empty string.
+            multiprocess_mode (str): The multiprocess mode of the metric. Defaults to ``all``. Available options
+                                     are (``all``, ``min``, ``max``, ``livesum``, ``liveall``)
+        """
+        return o
 
     @property
     def Info(self):
-        return partial(
-            self.prometheus_client.Info,
-            registry=self.registry,
-        )
+        logger.warning("Info WILL NOT WORK in multiprocess mode.")  # pragma: no cover
+        return partial(self.prometheus_client.Info, registry=self.registry)
 
     @property
     def Enum(self):
-        return partial(
-            self.prometheus_client.Enum,
-            registry=self.registry,
-        )
+        logger.warning("Enum WILL NOT WORK in multiprocess mode.")  # pragma: no cover
+        return partial(self.prometheus_client.Enum, registry=self.registry)
 
     @property
     def Metric(self):
-        return partial(
-            self.prometheus_client.Metric,
-            registry=self.registry,
-        )
+        """
+        A Metric family and its samples.
+
+        This is a base class to be used by instrumentation client. Custom collectors should use ``bentoml.metrics.metrics_core.GaugeMetricFamily``, ``bentoml.metrics.metrics_core.CounterMetricFamily``, ``bentoml.metrics.metrics_core.SummaryMetricFamily`` instead.
```

Comment:
Do we need to expose this then?

Reply:
We don't document this. But I think we can remove this section alltogether.